### ISSUE TITLE
check_list.cpp.py has a tiny issue when checking virtual functions

### ISSUE THREAD

**Issue by hushell** [shogun-toolbox/shogun#1505]
This issue happens when I write code like below. It will treat CMAPInference as an abstract class, but it is not. Then in the clone_equals unit-test, the class_list.cpp::new_sgserializable() returns NULL, and fail that test. 

class CMAPInference : CSGObject
{
    ...
    virtual void loss_augmentation(SGVector<int32_t> states_gt, 
        SGVector<float64_t> loss = SGVector<float64_t>());
    ...
};
class CMAPInferImpl : public CSGObject
{
    ...
    virtual float64_t inference(SGVector<int32_t> assignment) = 0;
};

So loss_augmentation() is not a virtual function in fact, but the test_candidate() in class_list.cpp.py will treat it as a virtual function, because it searches outside the block (i.e. code between '{' and '}') of CMAPInference to find if there is something like '= 0;'. In my case, it find the '= 0;' in another class CMAPInferImpl. 

It took me long time to find the problem, which seems comes from line 174 in class_list.cpp.py:
vstart,vstop=extract_block(c, lines, line_nr, len(lines), '(',')')
I think the 4th parameter 'len(lines)' should be 'stop', so that it will not check a line outside (start, stop) (see test_candidate() in class_list.cpp.py). This change works for me. 

@karlnapf @sonney2k Please let me know whether this is an issue.
